### DISCUSSION TITLE
How do you reflect a custom asset?

### DISCUSSION THREAD

**Discussion by juliohq** [bevyengine/bevy#15985] [Q&A] [answered]
In a custom resource you could do:
```rust
#[derive(Resource, Reflect, Default)]
#[reflect(Resource)]
MyCustomResource;
```
Though, it doesn't apply for assets:
```rust
#[derive(Asset, Reflect, Default)]
#[reflect(Asset) // not found in this scope
MyCustomAsset;
```
Is there something I'm missing or it isn't possible to reflect custom assets at all? I've been enjoying Bevy a lot so far!

**Comment by MrGVSV** [comment, marked as answer]
You need to import `bevy::asset::ReflectAsset`.

We should probably add this to the prelude, though, it doesn't seem to have much usage since it also requires special registration using `App::register_asset_reflect`.

**Comment by juliohq** [comment]
Yes it works, though I'm trying to make it possible to inspect my custom asset in bevy egui inspector but I'm still getting this error: `Handle<Map> has no ReflectHandle type data, so it cannot be displayed`. Is there anything I can do from my side or is that something the crate should do?

Edit: I fixed it by adding `app.register_asset_reflect::<Map>()`.
